lib/jekyll/layout.rb

```diff
@@ -43,7 +45,11 @@ def initialize(site, base, name)
     #
     # Returns nothing.
     def process(name)
-      self.ext = File.extname(name)
+      self.mime_type = MIME::Types.find{|t| ".#{t.preferred_extension}" == File.extname(name)}
```

Comment:
Perhaps an example: I've seen `application/vnd.geo+json` in both `.json` and `.geojson` files